perception/autoware_ground_segmentation/src/scan_ground_filter/node.cpp

```diff
@@ -336,239 +373,271 @@ void ScanGroundFilterComponent::classifyPointCloudGridScan(
   const PointCloud2ConstPtr & in_cloud, std::vector<PointCloudVector> & in_radial_ordered_clouds,
   pcl::PointIndices & out_no_ground_indices)
 {
-  out_no_ground_indices.indices.clear();
-  for (size_t i = 0; i < in_radial_ordered_clouds.size(); ++i) {
-    PointsCentroid ground_cluster;
-    ground_cluster.initialize();
-    std::vector<GridCenter> gnd_grids;
-    GridCenter curr_gnd_grid;
-
-    // check empty ray
-    if (in_radial_ordered_clouds[i].size() == 0) {
-      continue;
-    }
-
-    // check the first point in ray
-    auto * p = &in_radial_ordered_clouds[i][0];
-
-    bool initialized_first_gnd_grid = false;
-    bool prev_list_init = false;
-    pcl::PointXYZ p_orig_point, prev_p_orig_point;
-    for (auto & point : in_radial_ordered_clouds[i]) {
-      auto * prev_p = p;  // for checking the distance to prev point
-      prev_p_orig_point = p_orig_point;
-      p = &point;
-      get_point_from_global_offset(in_cloud, p_orig_point, in_cloud->point_step * p->orig_index);
-      float global_slope_ratio_p = p_orig_point.z / p->radius;
-      float non_ground_height_threshold_local = non_ground_height_threshold_;
-      if (p_orig_point.x < low_priority_region_x_) {
-        non_ground_height_threshold_local =
-          non_ground_height_threshold_ * abs(p_orig_point.x / low_priority_region_x_);
-      }
-      // classify first grid's point cloud
-      if (
-        !initialized_first_gnd_grid && global_slope_ratio_p >= global_slope_max_ratio_ &&
-        p_orig_point.z > non_ground_height_threshold_local) {
-        out_no_ground_indices.indices.push_back(p->orig_index);
-        p->point_state = PointLabel::NON_GROUND;
+  std::unique_ptr<ScopedTimeTrack> st_ptr;
+  if (time_keeper_) st_ptr = std::make_unique<ScopedTimeTrack>(__func__, *time_keeper_);
+
+  {  // add scope for time keeper (classify_loop)
```

Comment:
This scope is the only one in the function `classifyPointCloudGridScan`. therfore, the scope is not needed.